src/message_bus.zig

```diff
@@ -806,6 +806,7 @@ fn MessageBusType(comptime process_type: vsr.ProcessType) type {
                         log.info("connection from replica {}", .{connection.peer.replica});
                     },
                     .client => {
+                        assert(header.client != 0);
```

Comment:
This was already true before this PR. If we have determined that this connection is from a client, we expect that the header has an included client id.